jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassURLHelperImpl.java

```diff
@@ -47,12 +52,28 @@
 		initializeShareableClassloader(loader);
 	}
 	
-	private static native void init(); 
+	private static native void init();
+	
+	/* The passed-in URL path contains a jar file */
+	private boolean newJarFileCheck(URL convertedjarPath) {
+		boolean newJarFile = false;
+		String protocol = convertedjarPath.getProtocol();
+		String jarFile = convertedjarPath.getFile();
+		if ((protocol.equalsIgnoreCase("file") || protocol.equalsIgnoreCase("jar"))  //$NON-NLS-1$ //$NON-NLS-2$
+		&& jarFile.contains(".jar") //$NON-NLS-1$
+		) {
+			if (!jarFileNameCache.contains(jarFile)) {
```

Comment:
This needs to be thread safe. You can create a "ConcurrentHashSet" backed by a ConcurrentHashMap (see ConcurrentHashMap javadoc).

Reply:
Already fixed against the suggestion above.